From lethal@linux-sh.org Tue Feb 21 07:10:32 2006
Date: Tue, 21 Feb 2006 17:10:31 +0200
From: Paul Mundt <lethal@linux-sh.org>
To: Greg KH <greg@kroah.com>
Cc: <zanussi@us.ibm.com>
Subject: sysfs: Add __ATTR_RELAY() helper for relay attributes.
Message-ID: <20060221151031.GA20816@linux-sh.org>
Content-Disposition: inline

This adds a simple __ATTR_RELAY() to help people define relay attributes,
this takes care of things like getting the module owner right.

Signed-off-by: Paul Mundt <lethal@linux-sh.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/sysfs.h |   10 ++++++++++
 1 file changed, 10 insertions(+)

--- gregkh-2.6.orig/include/linux/sysfs.h
+++ gregkh-2.6/include/linux/sysfs.h
@@ -44,6 +44,16 @@ struct attribute_group {
 	.show	= _name##_show,	\
 }
 
+#define __ATTR_RELAY(_name,_buffer_size,_nr_buffers) {	\
+	.attr	= {					\
+		.owner = THIS_MODULE,			\
+		.name = __stringify(_name),		\
+		.mode = 0400,				\
+	},						\
+	.subbuf_size	= _buffer_size,			\
+	.n_subbufs	= _nr_buffers,			\
+}
+
 #define __ATTR_NULL { .attr = { .name = NULL } }
 
 #define attr_name(_attr) (_attr).attr.name
